merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -402,10 +403,10 @@ public SimulationResults computeResults(
       final var directive = engine.taskDirective.get(task);
       if (directive == null) return;
 
-      taskToPlannedDirective.computeIfAbsent(task.id(), id -> id);
+      taskToPlannedDirective.computeIfAbsent(task.id(), ActivityInstanceId::new);
```

Comment:
Somewhat relevant to this line:

> We should do the analogous work for plan IDs.

Would task IDs also be eligible for this change (for consistency across Merlin)?